src/couch_log/test/eunit/couch_log_formatter_test.erl

```diff
@@ -25,6 +25,22 @@ truncate_test() ->
     Entry = couch_log_formatter:format(info, self(), Msg),
     ?assert(length(Entry#log_entry.msg) =< 16000).
 
+format_report_etoolong_test() ->
+    Payload = lists:flatten(["a" || _ <- lists:seq(1, 1048576)]),
+    Resp = couch_log_formatter:format_report(self(), report123, #{
+        msg => Payload
+    }),
+    ?assertEqual({error, emsgtoolong}, Resp).
+
+format_report_test() ->
+    {ok, Entry} = couch_log_formatter:format_report(self(), report123, #{
+        foo => 123,
+        bar => "barStr",
+        baz => baz
+    }),
+    Formatted = "[foo=\"123\" baz=\"baz\" bar=\"barStr\"]",
```

Comment:
The expectation of a stable order of keys might not hold in future versions of OTP. The documentation for `maps:fold` used in `format_meta` states:

> Calls F(Key, Value, AccIn) for every Key to value Value association in MapOrIter in any order. 

I think we can accept the risk for a test case. However you can decide whether you want to handle it.

Reply:
@iilyak hmmm... so I specifically wanted to avoid needing to sort the keys in the output string and to rely upon a fixed formatting, but you're correct in pointing out that our test expects a consistent output.

Again, I don't want to sort the keys in the report for the normal use case just to get a specific test result. The "fix" that jumps out to me would be to dissect the resulting output string and assert the expected keys. For instance, we could first regex pattern match on `^[` and `]$` to assert the formatted output is in brackets. Then we could split the inner string on spaces, sort the output, and then assert each of the key/value pairs matches up. I'm somewhat inclined to leave this and see if we encounter it as a problem down the road, but I think the approach I just outlined is viable if folks want to avoid that eventuality. Alternative solutions welcome too 👍

Reply:
> The "fix" that jumps out to me would be to dissect the resulting output string and assert the expected keys. For instance, we could first regex pattern match on ^[ and ]$ to assert the formatted output is in brackets. Then we could split the inner string on spaces, sort the output, and then assert each of the key/value pairs matches up.

Something like

```erlang

parse_data(String) ->
   maps:from_list(
       lists:map(fun(X) -> 
           [K, V] = string:split(X, "="),
           {list_to_atom(K), V} 
       end, string:split(
           string:trim(String, both, [$[, $]]), 
           " ", 
           all))).
```

Another approach is based on the fact that we don't need to fully parse the data. For us it is enough to get to the level of

```
[
"bar=\"barStr\"",
"baz=\"baz\"",
"foo=\"123\""
]
``` 

It could be done as following

```erlang
Expected = split_and_order("[foo=\"123\" baz=\"baz\" bar=\"barStr\"]"),
?assertEqual(Expected, lists:flatten(split_and_order(Entry#log_entry.msg))),
?assertEqual($[, lists:first(Entry#log_entry.msg)).
?assertEqual($], lists:last(Entry#log_entry.msg)).


def split_and_order(String) -> 
    lists:sort(
         string:split(
             string:trim(String, both, [$[, $]]), 
             " ", 
             all
         )
    ).
```

Reply:
Thanks for the feedback. I agree the generic test version is quite a bit more complex and I'm not a huge fan. For now I've added a comment at the test location.